tools/getrealdeps.rb

```diff
@@ -1,30 +1,39 @@
 #!/usr/bin/env ruby
-# getrealdeps for Chromebrew version 1.2
+# getrealdeps for Chromebrew version 1.3
 # Author: Satadru Pramanik (satmandu) satadru at gmail dot com
 require 'fileutils'
 
+if ARGV.include?('--use-crew-dest-dir')
+  ARGV.delete('--use-crew-dest-dir')
+  @opt_use_crew_dest_dir = true
+end
+
 # Exit quickly if an invalid package name is given.
 if ARGV[0].nil? || ARGV[0].empty? || ARGV[0].include?('#')
-  puts 'Getrealdeps checks for the dependencies of a package.'
-  puts 'Dependencies are added if the package file is missing them.'
-  puts 'Usage: getrealdeps.rb <packagename>'
+  puts 'Getrealdeps checks for the runtime dependencies of a package.'
+  puts 'The runtime dependencies are added if the package file is missing them.'
+  puts 'Usage: getrealdeps.rb [--use_crew_dest_dir] <packagename>'
   exit 1
 end
 
 pkg = ARGV[0].chomp('.rb')
+puts "Checking for the runtime dependencies of #{pkg}..."
 
 CREW_PREFIX = ENV['CREW_PREFIX'] || `crew const CREW_PREFIX`.split('=')[1].chomp
+CREW_DEST_DIR = ENV['CREW_DEST_DIR'] || `crew const CREW_DEST_DIR`.split('=')[1].chomp
 CREW_LIB_PREFIX = ENV['CREW_LIB_PREFIX'] || `crew const CREW_LIB_PREFIX`.split('=')[1].chomp
 
-# Package needs to be installed for package filelist to be populated.
-unless File.exist? "#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist"
-  puts "Installing #{pkg} because it is not installed."
-  system("yes | crew install #{pkg}")
-end
-
-unless File.exist?("#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist")
-  puts "Package #{pkg} either does not exist or does not contain any libraries."
-  exit 1
+if @opt_use_crew_dest_dir
+  define_singleton_method('pkgfilelist') {File.join(CREW_DEST_DIR, 'filelist')}
+  abort('Pkg was not built.') unless File.exist?(pkgfilelist)
+else
+  define_singleton_method('pkgfilelist') {"#{CREW_PREFIX}/etc/crew/meta/#{pkg}.filelist"}
+  # Package needs to be installed for package filelist to be populated.
+  unless File.exist?(pkgfilelist)
+    puts "Installing #{pkg} because it is not installed."
+    system("yes | crew install #{pkg}")
```

Comment:
Check to see if local binary files exist and the version is updated prior to install?

Reply:
That code is for the case of `getrealdeps.rb` being run standalone. It isn't used in a build any more, since builds are shunted to `@opt_use_crew_dest_dir`